cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueClientFactory.java

```diff
@@ -84,6 +86,16 @@ public HBaseQueueClientFactory(CConfiguration cConf, Configuration hConf, HBaseT
   }
 
   @Override
+  public void initContext(Id.Run run) {
+    queueAdmin.initContext(run);
```

Comment:
The problem is not the queue admin, it is the `LineageWriterDatasetFramework` is not a singleton and it cannot be.